### PR TITLE
Fixes/als 790 tracking UI

### PR DESCRIPTION
**Pull request by zeeshanmakeen**
ALS-790: iOS - Enable tracking UI issues.
ALS-791: iOS - Tracking Empty state UI issues 
ALS-802: [iOS] Tracker and tracker history issues
ALS-764: [iOS] Explore, tracking, geofence, navigation should not be a modal rather a stand-alone menu

### CODE REVIEW

File: LocationServices/LocationServices/Extensions/UIViewController+Extension.swift [hunk 1]
```diff
@@ -17,4 +17,149 @@ extension UIViewController {
...
+    }
+    
+    func presentBottomSheet(parentController: UIViewController) {
+        parentController.addChild(self)
+        parentController.view.addSubview(self.view)
+        self.didMove(toParent: parentController)
+        self.view.snp.makeConstraints {
+            $0.leading.equalTo((parent?.view.snp.leading)!)
+            $0.trailing.equalTo((parent?.view.snp.trailing)!)
+            $0.bottom.equalTo((parent?.view.snp.bottom)!)
```
Comment (denis-ryzhkov-makeen) [inline LocationServices/LocationServices/Extensions/UIViewController+Extension.swift:52]: let's not use force unwrap

File: LocationServices/LocationServices/Views/Common/CommonDashboardView.swift [hunk 2]
```diff
@@ -95,17 +96,19 @@ final class CommonDashboardView: UIView {
...
         self.detailLabel.font = UIFont.amazonFont(type: .regular, size: 13)
         self.iconView.image = image
         self.iconView.backgroundColor = iconBackgroundColor
+        self.iconView.tintColor = .black
         self.comonButton.setTitle(buttonTitle, for: .normal)
     }
     
     func hideMaybeLaterButton(state: Bool) {
-        self.maybeLaterButton.isHidden = state
+        self.maybeLaterButton.isHidden = true // state
```
Comment (denis-ryzhkov-makeen) [inline LocationServices/LocationServices/Views/Common/CommonDashboardView.swift:111]: debug code?

**Review by denis-ryzhkov-makeen — CHANGES_REQUESTED** [review]
The general improvements that should be applied:
- when drag the view down it is shown over tabbar. Should slide under.
- when we show keyboard expand the view to full height. When we shrink view from full large to medium the keyboard should disappear. (This are embellishments that are working out of the box for native sheet controller. Please, communicate this with Stas if we need to implement them)
- after closing add geofence screen, the geofence list is presented even if I've not opened it
- can't close poi card
- check with Stas if scroll of scrollviews and tableviews inside viewcontroller should expand/interract bottom sheet

File: LocationServices/LocationServices/Extensions/UIViewController+Extension.swift [hunk 3]
```diff
@@ -17,4 +17,149 @@ extension UIViewController {
...
+    
+    func getMediumDetentHeight() -> CGFloat {
+        return self.parent!.view.frame.height * 0.5
+    }
+    
+    func getLargeDetentHeight() -> CGFloat {
+        return self.parent!.view.frame.height * 0.95
+    }
+    
+    func getBottomSheetHeightConstraint() -> [NSLayoutConstraint] {
```
[the remark this answers is not in the document]
Comment (zeeshanmakeen) [reply LocationServices/LocationServices/Extensions/UIViewController+Extension.swift:33]: this function is unused. removed

File: LocationServices/LocationServices/Coordinators/ChildCoordinators/TrackingCoordinator.swift [hunk 4]
```diff
@@ -37,61 +41,46 @@ extension TrackingCoordinator: TrackingNavigationDelegate {
...
-            sheet.detents = [.medium(), .large()]
-            sheet.prefersScrollingExpandsWhenScrolledToEdge = false
-            sheet.largestUndimmedDetentIdentifier = .medium
-            sheet.prefersGrabberVisible = true
-            sheet.preferredCornerRadius = 10
+            self?.currentBottomSheet?.view.removeFromSuperview()
+            if(self?.trackingController?.viewModel.isTrackingActive == true){
+                self?.showTrackingHistory(isTrackingActive: true)
+            }
+            else{
```
[the remark this answers is not in the document]
Comment (zeeshanmakeen) [reply LocationServices/LocationServices/Coordinators/ChildCoordinators/TrackingCoordinator.swift:76]: done

File: LocationServices/LocationServices/Coordinators/ChildCoordinators/TrackingCoordinator.swift [hunk 5]
```diff
@@ -145,15 +134,15 @@ extension TrackingCoordinator: TrackingNavigationDelegate {
...
             self.didSendEventClosure?()
         }
         
-        controller.directionHandler = {
+        trackingController!.directionHandler = {
             self.didSendDirectionEvent?()
         }
-        controller.delegate = self
-        navigationController.pushViewController(controller, animated: true)
+        trackingController!.delegate = self
```
[the remark this answers is not in the document]
Comment (zeeshanmakeen) [reply LocationServices/LocationServices/Coordinators/ChildCoordinators/TrackingCoordinator.swift:145]: done

File: LocationServices/LocationServices/Extensions/UIViewController+Extension.swift [hunk 6]
```diff
@@ -17,4 +17,149 @@ extension UIViewController {
...
+       grabberView.addGestureRecognizer(tapGestureRecognizer)
+    }
+    
+    func updateBottomSheetHeight(to height: CGFloat) {
+        self.view.snp.updateConstraints{
+            $0.height.equalTo(height)
+        }
+    }
+    
+    func setBottomSheetHeight(to height: CGFloat) {
```
[the remark this answers is not in the document]
Comment (zeeshanmakeen) [reply LocationServices/LocationServices/Extensions/UIViewController+Extension.swift:83]: I think the short name is sufficient. Also, it is not the scope of this function to remove gesture. We can have another function but that is not needed anywhere right now.

File: LocationServices/LocationServices/Extensions/UIViewController+Extension.swift [hunk 7]
```diff
@@ -17,4 +17,149 @@ extension UIViewController {
...
+        }
+    }
+
+    @objc private func handlePanGesture(_ recognizer: UIPanGestureRecognizer) {
+        let translation = recognizer.translation(in: self.view)
+        let currentHeight = view.frame.height - translation.y
+        
+        let newHeight = max(min(currentHeight, getLargeDetentHeight()), getSmallDetentHeight())
+
+        if recognizer.state == .changed {
```
[the remark this answers is not in the document]
Comment (zeeshanmakeen) [reply LocationServices/LocationServices/Extensions/UIViewController+Extension.swift:95]: changed & refactored

File: LocationServices/LocationServices/Extensions/UIViewController+Extension.swift (hunk 1 above)
Comment (zeeshanmakeen) [reply LocationServices/LocationServices/Extensions/UIViewController+Extension.swift:52]: done

File: LocationServices/LocationServices/Views/Common/CommonDashboardView.swift (hunk 2 above)
Comment (zeeshanmakeen) [reply LocationServices/LocationServices/Views/Common/CommonDashboardView.swift:111]: No it is asked not to show maybe later button. I've not removed it but just hid it

File: LocationServices/LocationServices/Extensions/UIViewController+Extension.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (zeeshanmakeen) [reply LocationServices/LocationServices/Extensions/UIViewController+Extension.swift:33]: this function is unused. removed

File: LocationServices/LocationServices/Coordinators/ChildCoordinators/TrackingCoordinator.swift (hunk 5 above)
[the remark this answers is not in the document]
Comment (zeeshanmakeen) [reply LocationServices/LocationServices/Coordinators/ChildCoordinators/TrackingCoordinator.swift]: done

> The general improvements that should be applied:
> 
> * when drag the view down it is shown over tabbar. Should slide under.
> * when we show keyboard expand the view to full height. When we shrink view from full large to medium the keyboard should disappear. (This are embellishments that are working out of the box for native sheet controller. Please, communicate this with Stas if we need to implement them)
> * after closing add geofence screen, the geofence list is presented even if I've not opened it
> * can't close poi card
> * check with Stas if scroll of scrollviews and tableviews inside viewcontroller should expand/interract bottom sheet

All Done

[the remark this answers is not in the document]
Comment (zeeshanmakeen) [reply LocationServices/LocationServices/Coordinators/ChildCoordinators/TrackingCoordinator.swift]: Will ask for review from stas once it is approved and build is provided

File: LocationServices/LocationServices/Views/Common/CommonDashboardView.swift (hunk 2 above)
Comment (zeeshanmakeen) [reply LocationServices/LocationServices/Views/Common/CommonDashboardView.swift:111]: No it is asked not to show maybe later button. I've not removed it but just hid it

**Pull request merged**